geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -56,29 +57,21 @@ trait IngestCommand[DS <: DataStore]
 
   override def execute(): Unit = {
     if (params.files.isEmpty && !StdInHandle.isAvailable) {
-      throw new ParameterException("Missing option: <files>... is required")
+      throw new ParameterException("Missing option: <files>... is required, or use `-` to ingest from standard in")
     }
 
-    val inputs = if (params.srcList) {
-      val lists = if (params.files.isEmpty) { StdInHandle.available().toList } else {
-        params.files.asScala.flatMap(PathUtils.interpretPath).toList
-      }
-      lists.flatMap { file =>
-        WithClose(file.open) { iter =>
-          iter.flatMap { case (_, is) => IOUtils.lineIterator(is, StandardCharsets.UTF_8).asScala }.toList
-        }
-      }
-    } else {
-      params.files.asScala
+    val inputs: Inputs = {
+      val files = Inputs(params.files.asScala)
+      if (params.srcList) { files.asSourceList } else { files }
```

Comment:
Wow, let me make sure I've got this...  One could pass in via stdin data or a list of files to ingest?  Neat.

Reply:
yeah, i think you could do this before too